pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -116,6 +116,18 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 		return reconcile.Result{}, fmt.Errorf("errors: %v %v", err, updateErr)
 	}
 
+	if buildRun.Labels == nil {
+		buildRun.Labels = make(map[string]string)
```

Comment:
are labels not already of that type, why you need to redefine them again?

Reply:
If there is no label setting in the resource, it is nil, if you want to add something in the label, kube will report null point error

We say this kind of problem for both annotation and lable in our PoC

Reply:
u re right!

Reply:
also you dont need the make , you can always do `map[string]string{}` is simpler, just a tip!